### PR TITLE
TST: moved file test_concat.py to folder ./concat/ (#37243)

### PR DESCRIPTION
**Pull request by kamilt5**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/reshape/concat/test_concat.py [hunk 1]
```diff
@@ -1845,124 +1845,6 @@ def test_concat_preserves_extension_int64_dtype():
     tm.assert_frame_equal(result, expected)
```
Comment (laffite56) [inline pandas/tests/reshape/concat/test_concat.py]: There is typo in commit message, it should be "split class TestSeriesConcat to file test_series.py"

**Comment by laffite56** [comment]
so after merging this PR to master, I will be able to create new PR and do follows-up? Further splitting to "test_categorial", "test_empty" require moving methods, so doing this will be better in another PR, yes?

**Comment by jreback** [comment]
ok, need you to run the pre-commit hooks: https://github.com/pandas-dev/pandas/pull/37360/checks?check_run_id=1297616903

and ping on green and can merge (then can do followups).

**Comment by jreback** [comment]
great ping on green.

**Comment by laffite56** [comment]
Ugh, why did it crash with: 
**AssertionError: ('pandas/tests/io/parser/test_c_parser_only.py::test_parse_trim_buffers[c_low]', <WorkerController gw0>)**

Is it connected with my commit?

**Comment by jreback** [comment]
thanks @laffite56 

no the windows checks have been finicky lately.

pls issue followon PRs to clean this up.

**Pull request merged**